common/unsafe/src/main/java/org/apache/spark/unsafe/memory/HeapMemoryAllocator.java

```diff
@@ -73,6 +73,12 @@ public MemoryBlock allocate(long size) throws OutOfMemoryError {
 
   @Override
   public void free(MemoryBlock memory) {
+    assert(memory instanceof ByteArrayMemoryBlock || memory instanceof IntArrayMemoryBlock ||
```

Comment:
we expect to only see `LongArrayMemoryBlock` here, don't we?

Reply:
Good catch, you are right.